docs/pages/web/banneroverlay.js

```diff
@@ -0,0 +1,226 @@
+import image from '../../examples/banneroverlay/image';
+import main from '../../examples/banneroverlay/main';
+import mobile from '../../examples/banneroverlay/mobile';
+
+export default function DocsPage({ generatedDocGen }: { generatedDocGen: DocGen }): ReactNode {
+  return (
+    <Page title={generatedDocGen?.displayName}>
+      <PageHeader
+        name={generatedDocGen?.displayName}
+        description={generatedDocGen?.description}
+        pdocsLink
+      >
+        <SandpackExample code={main} name="Main BannerOverlay example" hideEditor />
+      </PageHeader>
+      <GeneratedPropTable generatedDocGen={generatedDocGen} />
+      <MainSection name="Usage guidelines">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            title="When to use"
+            description={`
+- To provide short educational messages allowing content to scroll underneath
+- To help users learn more about a specific idea or interest
+- To support users when they have performed actions that indicate medium or high intent (related pin tap, idea pin swipe)
+        `}
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            title="When not to use"
+            description={`
+- If there is a need to block the content underneath use Modal instead
+- To replace Toast
+- For lengthy messages, forms, or blocks of information. Consider OverlayPanel or new page instead
+        `}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <MainSection name="Best Practices">
+        <MainSection.Subsection columns={1}>
+          <MainSection.Card
+            type="do"
+            sandpackExample={<SandpackExample code={doEducate} name="Do - Educate" hideEditor />}
+            description="Use BannerOverlay to educate and provide additional information to users when they have performed actions that indicate some intent, such as a related pin tap or idea pin swipe."
+          />
+          <MainSection.Card
+            type="don't"
+            sandpackExample={
+              <SandpackExample
+                code={dontCritical}
+                name="Don't - Critical"
+                hideEditor
+                hideControls
+              />
+            }
+            description="Use BannerOverlay for critical information, such as errors or warnings. Use Callout instead."
+          />
+          <MainSection.Card
+            type="do"
+            sandpackExample={<SandpackExample code={doConcise} name="Do - Concise" hideEditor />}
+            description="Be concise when writing the content. The BannerOverlay is intended to display short messages. Ideally max of 3 lines.
+Please consider localization. "
+          />
+          <MainSection.Card
+            type="don't"
+            sandpackExample={
+              <SandpackExample
+                code={dontLong}
```

Comment:
Hey in this example:

<img width="932" alt="image" src="https://github.com/pinterest/gestalt/assets/5509813/14c0e378-57b0-4359-b88d-47426154423a">

Is there a way to not mess-up the button with a min width on that area or something? 

Let's say in the bad case there's some poor localization and someone does do this. This can create a breaking experience. Maybe we should grow the container size, or wrap the text into new lines. wdyt?

Reply:
Also, I get it's a don't example lol, so we don't have to do this

Reply:
Added a nowrap style to the CTA, to prevent this from happening :)